### ISSUE TITLE
Stack limit Exceeded when using gestures in ionic 2

### ISSUE THREAD

**Issue by muhammedabuali** [driftyco/ionic#5663]
#### Short description of the problem:
when a gesture is triggered in ionic 2, the browser console shows a stack limit exceeded error

#### What behavior are you expecting?
the corresponding function should be triggered and there shouldn't be an error

**Steps to reproduce:**
1. add a gesture to a HTML element like `(swipe)` or `(tap)`
2. fire the event by taping on the item
3. watch the error gets thrown in the browser console

**Ionic Version**: 1.x / 2.x
version 2.0.0-beta.17
**Browser & Operating System**: iOS / Android / Chrome
Chrome

**Run `ionic info` from terminal/cmd prompt:**

```
Cordova CLI: 5.4.1
Ionic CLI Version: 2.0.0-beta.17
Ionic App Lib Version: 2.0.0-beta.8
OS: Distributor ID:	Fedora Description:	Fedora release 23 (Twenty Three) 
Node Version: v4.3.1
```

**Comment by perrygovier** [comment]
Please add Dgeni docs regarding gestures when working on this

**Comment by adamdbradley** [comment]
Ng2 issue: https://github.com/angular/angular/issues/6993

**Comment by astec** [comment]
Should be fixed by https://github.com/angular/angular/commit/41e38e4330bee5f117d823aa51204449650cb385?

**Comment by godzsa** [comment]
This is an angular2 bug. 
https://github.com/angular/angular/issues/6993#issuecomment-187115690
I guess it will be fixed shortly.

**Comment by phoenixgao** [comment]
This is a bug in angular2, at the time of this comment it's fixed but not released (would be released with angular2: 2.0.0-beta10). So for now you can fix it manually by update your angular2:

`$ npm install git+ssh://git@github.com:angular/angular.git#builds-js`

**Comment by adamdbradley** [comment]
Fixed in ng2 and will be released in beta4
